databricks/koalas/series.py

```diff
@@ -1451,6 +1451,154 @@ def clip(self, lower: Union[float, int] = None, upper: Union[float, int] = None)
+        >>> s = ks.Series([45, 200, 1.2, 30, 250, 1.5, 320, 1, 0.3],
+        ...               index=midx)
+        >>> s
+        lama    speed      45.0
+                weight    200.0
+                length      1.2
+        cow     speed      30.0
+                weight    250.0
+                length      1.5
+        falcon  speed     320.0
+                weight      1.0
+                length      0.3
+        Name: 0, dtype: float64
+
+        >>> s.drop(labels='weight', level=1)
+        lama    speed      45.0
+                length      1.2
+        cow     speed      30.0
+                length      1.5
+        falcon  speed     320.0
+                length      0.3
+        Name: 0, dtype: float64
+
+        >>> s.drop(('lama', 'weight'))
+        lama    speed      45.0
+                length      1.2
+        cow     speed      30.0
+                weight    250.0
+                length      1.5
+        falcon  speed     320.0
+                weight      1.0
+                length      0.3
+        Name: 0, dtype: float64
+
+        >>> s.drop([('lama', 'speed'), ('falcon', 'weight')])
+        lama    weight    200.0
+                length      1.2
+        cow     speed      30.0
+                weight    250.0
+                length      1.5
+        falcon  speed     320.0
+                length      0.3
+        Name: 0, dtype: float64
+        """
+        if labels is not None:
+            if index is not None:
+                raise ValueError("Cannot specify both 'labels' and 'index'")
+            return self.drop(index=labels, level=level)
+        if index is not None:
+            if not isinstance(index, (str, tuple, list)):
+                raise ValueError("'index' type should be one of str, list, tuple")
+            if level is None:
+                level = 0
+            if level >= len(self._internal.index_scols):
+                raise ValueError("'level' should be less than the number of indexes")
+
+            if isinstance(index, str):
+                index = [(index,)]  # type: ignore
+            elif isinstance(index, tuple):
+                index = [index]
+            else:
+                if not (all((isinstance(idxes, str) for idxes in index)) or
+                        all((isinstance(idxes, tuple) for idxes in index))):
+                    raise ValueError("If the given index is a list, it "
+                                     "should only contains names as strings, "
+                                     "or a list of tuples that contain "
+                                     "index names as strings")
+                index = [idxes if isinstance(idxes, tuple) else (idxes,)  # type: ignore
+                         for idxes in index]
+
+            drop_index_scols = []
+            for idxes in index:
+                index_scols = [self._internal.index_scols[lvl] == idx
+                               for lvl, idx in enumerate(idxes, level)]
```

Comment:
Maybe we should consider one more case:

with pandas:

```py
>>> s.drop(('weight', 'x', 'y'), level=1)
lama    speed      45.0
        weight    200.0
        length      1.2
cow     speed      30.0
        weight    250.0
        length      1.5
falcon  speed     320.0
        weight      1.0
        length      0.3
dtype: float64
```

whereas:

```py
>>> s.drop(('weight', 'x', 'y'), level=1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/ueshin/workspace/databricks-koalas/master/databricks/koalas/series.py", line 1568, in drop
    return self.drop(index=labels, level=level)
  File "/Users/ueshin/workspace/databricks-koalas/master/databricks/koalas/series.py", line 1594, in drop
    for lvl, idx in enumerate(idxes, level)]
  File "/Users/ueshin/workspace/databricks-koalas/master/databricks/koalas/series.py", line 1594, in <listcomp>
    for lvl, idx in enumerate(idxes, level)]
IndexError: list index out of range
```

I don't think the pandas behavior is right, but we might need to follow, or at least we should raise an error with a proper message instead of "index out of range".

Reply:
oh i got it. going to work right now :)

Reply:
i checked some cases,

They raises `KeyError` without specify level explicitly like below:

```
>>> s.toPandas().drop(('weight', 'x', 'y'))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/haejoonlee/Desktop/python/venv_3.7_selenium/lib/python3.7/site-packages/pandas/core/series.py", line 4321, in drop
    errors=errors,
  File "/Users/haejoonlee/Desktop/python/venv_3.7_selenium/lib/python3.7/site-packages/pandas/core/generic.py", line 3915, in drop
    obj = obj._drop_axis(labels, axis, level=level, errors=errors)
  File "/Users/haejoonlee/Desktop/python/venv_3.7_selenium/lib/python3.7/site-packages/pandas/core/generic.py", line 3947, in _drop_axis
    new_axis = axis.drop(labels, errors=errors)
  File "/Users/haejoonlee/Desktop/python/venv_3.7_selenium/lib/python3.7/site-packages/pandas/util/_decorators.py", line 208, in wrapper
    return func(*args, **kwargs)
  File "/Users/haejoonlee/Desktop/python/venv_3.7_selenium/lib/python3.7/site-packages/pandas/core/indexes/multi.py", line 2143, in drop
    loc = self.get_loc(level_codes)
  File "/Users/haejoonlee/Desktop/python/venv_3.7_selenium/lib/python3.7/site-packages/pandas/core/indexes/multi.py", line 2672, in get_loc
    "".format(keylen, self.nlevels)
KeyError: 'Key length (3) exceeds index depth (2)'
```

but when specify level explicitly, they just return series itself as is like you tried.

so i fixed them act exactly same with pandas.

P.S. As you said, I don't think the pandas behavior is right either. i think maybe both of them should raises key error